includes/ConvertToBlocks/Settings.php

```diff
@@ -0,0 +1,260 @@
+	 */
+	public function can_register() {
+		return is_admin();
+	}
+
+	/**
+	 * Configures variables and fetches post types.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		// Configure variables.
+		$this->settings_page    = sprintf( '%s-%s', CONVERT_TO_BLOCKS_SLUG, $this->settings_page );
+		$this->settings_section = sprintf( '%s-%s', CONVERT_TO_BLOCKS_SLUG, $this->settings_section );
+		$this->settings_group   = sprintf( '%s_%s', CONVERT_TO_BLOCKS_SLUG, $this->settings_group );
+
+		// Get post types.
+		$this->post_types = $this->get_post_types();
+	}
+
+	/**
+	 * Retrieves all public post types.
+	 *
+	 * @return array
+	 */
+	public function get_post_types() {
+		$post_types = get_post_types(
+			[ 'public' => true ]
+		);
+
+		if ( ! empty( $post_types['attachment'] ) ) {
+			unset( $post_types['attachment'] );
+		}
+
+		return $post_types;
+	}
+
+	/**
+	 * Adds a submenu item for the `Settings` menu.
+	 *
+	 * @return void
+	 */
+	public function add_menu() {
+		add_options_page(
+			esc_html__( 'Convert to Blocks', 'convert-to-blocks' ),
+			esc_html__( 'Convert to Blocks', 'convert-to-blocks' ),
+			$this->capability,
+			CONVERT_TO_BLOCKS_SLUG,
+			[ $this, 'settings_page' ]
+		);
+	}
+
+	/**
+	 * Registers the settings page.
+	 *
+	 * @return void
+	 */
+	public function settings_page() {
+		?>
+		<div class="wrap">
+			<h1>
+				<?php esc_html_e( 'Convert to Blocks', 'convert-to-blocks' ); ?>
+			</h1>
+			<hr>
+
+			<p>
+				<?php esc_html_e( 'Configure plugin by selecting the supported post types.', 'convert-to-blocks' ); ?>
+			</p>
+
+			<form method="post" action="options.php">
+				<?php
+				settings_fields( $this->settings_group );
+				do_settings_sections( CONVERT_TO_BLOCKS_SLUG );
+				submit_button();
+				?>
+			</form>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Registers section for the settings page.
+	 *
+	 * @return void
+	 */
+	public function register_section() {
+		add_settings_section(
+			$this->settings_section,
+			false,
+			false,
+			CONVERT_TO_BLOCKS_SLUG
+		);
+	}
+
+	/**
+	 * Registers setting fields.
+	 *
+	 * @return void
+	 */
+	public function register_fields() {
+		// Supported post types.
+		add_settings_field(
+			sprintf( '%s_post_types', CONVERT_TO_BLOCKS_SLUG ),
+			esc_html__( 'Supported Post Types', 'convert-to-blocks' ),
+			[ $this, 'field_post_types' ],
+			CONVERT_TO_BLOCKS_SLUG,
+			$this->settings_section,
+			[
+				'label_for' => sprintf( '%s_post_types', CONVERT_TO_BLOCKS_SLUG ),
+			]
+		);
+
+		register_setting(
+			$this->settings_group,
+			sprintf( '%s_post_types', CONVERT_TO_BLOCKS_SLUG ),
+			[
+				'sanitize_callback' => [ $this, 'sanitize_post_types' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders the `post_types` field.
+	 *
+	 * @return void
+	 */
+	public function field_post_types() {
+		$post_types  = get_option(
+			sprintf( '%s_post_types', CONVERT_TO_BLOCKS_SLUG ),
+			apply_filters( 'convert_to_blocks_default_post_types', CONVERT_TO_BLOCKS_DEFAULT_POST_TYPES )
```

Comment:
We should avoid using the same filter in different places when possible.

For this case, we can define the `$cointainer` property for this class, and get the default post types using `$this->container->get_default_post_types(), see [this](https://github.com/10up/convert-to-blocks/blob/d33e248866170f2d0372b72e83d8340b1eb13d11/includes/ConvertToBlocks/Plugin.php#L148-L157) for more details.

By doing so, we don't need the `CONVERT_TO_BLOCKS_DEFAULT_POST_TYPES` constant anymore.